clang/lib/Tooling/DependencyScanning/ModuleDepCollector.cpp

```diff
@@ -956,9 +956,12 @@ ModuleDepCollectorPP::handleTopLevelModule(const Module *M) {
   // Finish the compiler invocation. Requires dependencies and the context hash.
   MDC.addOutputPaths(CI, MD);
 
+  if (MDC.TransformInvocation)
+    MDC.TransformInvocation(CI);
```

Comment:
This should definitely run before `associateWithContextHash(CI, ...)`, `areOptionsInStableDir(CI, ...)` and `createCompileJobCacheKey(..., CI)`. What kind of transformation would clients typically perform?